tests/src/Interop/PInvoke/Miscellaneous/HandleRef/HandleRefNative.cpp

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <stdio.h>
+#include <xplatform.h>
+
+const int intManaged = 1000;
+const int intNative = 2000;
+const int intReturn = 3000;
+const int intErrReturn = 4000;
+const int expectedStackGuard = 5000;
```

Comment:
What is this supposed to do? I am unclear why this is being used or what it is actually checking/preventing?

Reply:
AFAICT it's there because `HandleRef` has a weird way of being marshalled. It is a pointer + an object that defines its lifetime, but only the pointer is marshalled. I think `expectedStackGuard` is there to basically ensure that we don't pass the object and end up incorrectly placing the rest of the parameters on the stack/in registers for the native call (which would happen if `HandleRef` marshalling accidentally marshalled the lifetime-defining member).